src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/WorkTypesController.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.controllers;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.slf4j.Logger;
+import org.springframework.http.CacheControl;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.access.AccessControlService;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.access.entities.AccessControlResponse;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.role.entities.RoleAssignment;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.task.WorkType;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.TaskManagementService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.slf4j.LoggerFactory.getLogger;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.config.SecurityConfiguration.AUTHORIZATION;
+
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis"})
+@RequestMapping(path = "/work-types", produces = APPLICATION_JSON_VALUE)
+@RestController
+public class WorkTypesController extends BaseController {
+    private static final Logger LOG = getLogger(WorkTypesController.class);
+
+    private final TaskManagementService taskManagementService;
+    private final AccessControlService accessControlService;
+
+    public WorkTypesController(TaskManagementService taskManagementService,
+                               AccessControlService accessControlService) {
+        super();
+        this.taskManagementService = taskManagementService;
+        this.accessControlService = accessControlService;
+    }
+
+    @ApiOperation("Retrieve a list of work types for the current user")
+    @ApiResponses({
+        @ApiResponse(code = 200, message = OK, response = String.class)
+    })
+    @GetMapping(path = "/users")
+    public ResponseEntity<List<WorkType>> getWorkTypes(@RequestHeader(AUTHORIZATION) String authToken) {
+
+        List<WorkType> workTypes = new ArrayList<>();
+
+        AccessControlResponse roles = accessControlService.getRoles(authToken);
+
+        LOG.info("Access Control Response  found [{}]", roles);
+
+        if (roles.getRoleAssignments().isEmpty()) {
+            return ResponseEntity
+                .ok()
+                .cacheControl(CacheControl.noCache())
+                .body(Collections.emptyList());
```

Comment:
I believe the access control would take care of this and throw an exception NoRoleAssignmentsFoundException so no need to handle it here as this code will never be reached

Reply:
resolved